core/src/main/java/com/spotify/ffwd/output/CoreOutputManager.java

```diff
@@ -181,6 +194,27 @@ public final Long getHLLPRefreshPeriodLimit() {
         } else {
             this.cardinalityLimit = null;
         }
+
+        if (!Strings.isNullOrEmpty(dynamicTagsFile)) {
+            final File file = new File(dynamicTagsFile);
+            if (file.exists() && file.canRead()) {
+                tagRefreshingThread = Executors.newSingleThreadScheduledExecutor(
+                    new ThreadFactoryBuilder().setNameFormat("tag-refreshing-thread").build());
+                tagRefreshingThread.scheduleWithFixedDelay(() -> refreshTagsFromFile(file.toPath()),
+                    1, 1, TimeUnit.MINUTES);
+            }
+        }
+    }
+
+    private void refreshTagsFromFile(final Path path) {
+        try (final Stream<String> lines = Files.lines(path)) {
+            lines.forEach(label -> {
+                final String[] labelParts = label.split("=", 2);
+                tags.put(labelParts[0], labelParts[1]);
+            });
+        } catch (IOException e) {
+            throw new RuntimeException(e);
```

Comment:
What should happen if the file isn't readable?

Reply:
This is where I was thinking maybe we fall back to the previous functionality or something along those lines

Reply:
So just log a warning and ignore?

Reply:
done